src/pages/Detail/Detail.js

```diff
@@ -0,0 +1,152 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { Link, useParams } from 'react-router-dom'
+
+import pokemonService from 'services/pokemonService'
+import { ErrorFetchDataContext } from 'contexts/ErrorFetchDataContextProvider'
+
+import AbilityOfPokemon from './components/AbilityOfPokemon'
+import VarietyOfPokemon from './components/VarietyOfPokemon'
+import './styles.css'
+
+export default function Detail() {
+  const params = useParams()
+
+  const id = params.id || 1
+
+  const { handleErrorFetchData } = useContext(ErrorFetchDataContext)
+
+  // save state detail pokemon.
+  const [detailPokemon, setDetailPokemon] = useState(() => ({
+    id: '',
+    name: '',
+    frontDefault: '',
+    types: '',
+    baseExperience: '',
+    weight: '',
+    height: '',
+    abilities: [],
+    stats: [],
+    speciesUrl: null
+  }))
+
+  const [evolutionChain, setEvolutionChain] = useState(() => ({}))
+
+  useEffect(() => {
+    if (detailPokemon.speciesUrl !== null) {
+      pokemonService
+        .getEvolutionChain(detailPokemon.speciesUrl)
+        .then(data => {
+          setEvolutionChain(data)
+        })
+        .catch(() => {
+          handleErrorFetchData()
+        })
+    }
+  }, [detailPokemon])
+
+  useEffect(() => {
+    pokemonService
+      .getOnePokemon(id)
+      .then(data => {
+        setDetailPokemon(data)
+      })
+      .catch(() => {
+        handleErrorFetchData()
+      })
+  }, [id])
+
+  return (
+    <div className="container mb-4">
+      {/* Logo */}
+      <h1 className="text-center mb-5" id="logo-page">
+        Pokemon Dex
+      </h1>
+
+      <div className="row mb-5">
+        <div className="col-md-6">
+          <img
+            src={`${detailPokemon.frontDefault}`}
+            className="img-fluid"
+            alt={detailPokemon.name}
+            width="100%"
+            height="100%"
+          />
+        </div>
+        <div className="col-md-6 pts-content-b">
+          <h3 className="title">Bio</h3>
+          <p>
+            <b>Name:</b> {`${detailPokemon.name}`.toUpperCase()}
+          </p>
+          <p>
+            <b>Types: </b>
+            {detailPokemon.types}
+          </p>
+          <p>
+            <b>Base experience:</b> {detailPokemon.baseExperience}
+          </p>
+          <p>
+            <span>
+              <b>Weight:</b> {detailPokemon.weight}kg
+            </span>
+            {` `}
+            <span>
+              <b>Height:</b> {detailPokemon.height}m
```

Comment:
please carefully take a look at pokemon api for pokemon details. It said that pokemon weight unit is not `kg`. Same for height unit